src/libbson/libbson.rc.in

```diff
@@ -28,11 +31,11 @@ BEGIN
             VALUE "CompanyName", "MongoDB, Inc"
             VALUE "FileDescription", "A BSON Library for C"
             VALUE "FileVersion", BSON_VERSION_S
-            VALUE "InternalName", "bson-1.0"
-            VALUE "OriginalFilename", "bson-1.0.dll"
+            VALUE "InternalName", "bson-" BSON_API_VERSION
```

Comment:
Should the InternalName be simply "bson" (no version number) similar to how it is simply "bsoncxx" in the CXX Driver?

Conversely I wonder if it should just be equal to OriginalFilename [per the field description](https://learn.microsoft.com/en-us/windows/win32/menurc/versioninfo-resource) (I don't think we have a notion of an "internal name" in the first place?):

> Internal name of the file, if one exists—for example, a module name if the file is a dynamic-link library. If the file has no internal name, this string should be the original filename, without extension. This string is required.

From quick reading of related documentation, I think the intent is for it to be set to the name of a "module" for languages that support modules (i.e. "module_name" given `import package.library.module_name`), which I don't think applies to our circumstances, at least not directly.

Reply:
I do not think the driver has a direct notion of internal name. Updated to match `OriginalFilename` without the extension. 

Will apply same update to C++ driver PR.